nix/wlroots.nix

```diff
@@ -1,64 +1,32 @@
 {
+  lib,
   version,
   src,
-  #
   wlroots,
-  xwayland,
-  fetchpatch,
-  lib,
   hwdata,
-  libliftoff,
   libdisplay-info,
-  hidpiXWayland ? true,
-  enableXWayland ? true,
-  nvidiaPatches ? false,
+  libliftoff,
+  enableNvidiaPatches ? false,
 }:
-assert (lib.assertMsg (hidpiXWayland -> enableXWayland) ''
-  wlroots-hyprland: cannot have hidpiXWayland when enableXWayland is false.
-'');
-  (wlroots.overrideAttrs
-    (old: {
-      inherit version src;
-      pname =
-        old.pname
-        + "-hyprland"
-        + (
-          if hidpiXWayland
-          then "-hidpi"
-          else ""
-        )
-        + (
-          if nvidiaPatches
-          then "-nvidia"
-          else ""
-        );
-      patches =
-        (old.patches or [])
-        ++ (lib.optionals (enableXWayland && hidpiXWayland) [
-          # adapted from https://gitlab.freedesktop.org/lilydjwg/wlroots/-/commit/6c5ffcd1fee9e44780a6a8792f74ecfbe24a1ca7
-          ./patches/wlroots-hidpi.patch
-          (fetchpatch {
-            url = "https://gitlab.freedesktop.org/wlroots/wlroots/-/commit/18595000f3a21502fd60bf213122859cc348f9af.diff";
-            sha256 = "sha256-jvfkAMh3gzkfuoRhB4E9T5X1Hu62wgUjj4tZkJm0mrI=";
-            revert = true;
-          })
-        ])
-        ++ (lib.optionals nvidiaPatches [
-          ./patches/wlroots-nvidia.patch
-        ]);
-      buildInputs = old.buildInputs ++ [hwdata libliftoff libdisplay-info];
+wlroots.overrideAttrs (old: {
+  inherit version src;
+
+  pname =
```

Comment:
Because of Alejandra, I prefer: `"${old.pname}-hyprland${lib.optionalString enableNvidiaPatches "-nvidia"}"`

Reply:
Hopefully nixfmt gets blessed as *the* formatter soon, so we won't have to worry.
In the meantime, I'll change it.

Reply:
> Hopefully nixfmt gets blessed as the formatter soon

You could help with that by using it :smile:

Reply:
Yes, too bad [the PR](https://github.com/serokell/nixfmt/pull/118) is still a draft.